cmd/completion/main.go

```diff
@@ -57,56 +63,112 @@ func init() {
+		log.Fatal(err, "toml decode")
 	}
 
-	logger.Println("Build successful")
-}
+	if len(report.Image.Tags) == 0 {
+		log.Fatal(errors.New("no image found in report"))
+	}
 
-func signImage() error {
-	var report platform.ExportReport
-	_, err := toml.DecodeFile(reportFilePath, &report)
+	builtImageRef := fmt.Sprintf("%s@%s", report.Image.Tags[0], report.Image.Digest)
+
+	k8sNodeKeychain, err := k8schain.NewNoClient(context.Background())
 	if err != nil {
-		return errors.Wrap(err, "toml decode")
+		log.Fatal(err)
 	}
 
-	creds, err := dockercreds.ParseMountedAnnotatedSecrets(registrySecretsDir, dockerCredentials)
+	creds, err := dockercreds.ParseBasicAuthSecrets(registrySecretsDir, dockerCredentials)
 	if err != nil {
-		return err
+		log.Fatal(err)
 	}
 
 	for _, c := range append(dockerCfgCredentials, dockerConfigCredentials...) {
 		credPath := filepath.Join(registrySecretsDir, c)
 
-		dockerCfgCreds, err := dockercreds.ParseDockerPullSecrets(credPath)
+		dockerConfigCreds, err := dockercreds.ParseDockerConfigSecret(credPath)
 		if err != nil {
-			return err
-		}
-
-		for domain := range dockerCfgCreds {
-			logger.Printf("Loading secret for %q from secret %q at location %q", domain, c, credPath)
+			log.Fatal(err)
 		}
 
-		creds, err = creds.Append(dockerCfgCreds)
+		creds, err = creds.Append(dockerConfigCreds)
 		if err != nil {
-			return err
+			log.Fatal(err)
 		}
+	}
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		log.Fatal(errors.Wrapf(err, "error obtaining home directory"))
+	}
+
+	err = creds.Save(filepath.Join(homeDir, ".docker", "config.json"))
+	if err != nil {
+		log.Fatal(errors.Wrapf(err, "error writing docker creds"))
+	}
+
+	keychain := authn.NewMultiKeychain(k8sNodeKeychain, creds)
+
+	metadataRetriever := cnb.RemoteMetadataRetriever{
+		Keychain:     keychain,
+		ImageFetcher: &registry.Client{},
+	}
 
-		homeDir, err := os.UserHomeDir()
+	cacheImageRef := ""
+	if cacheTag != "" {
+		cacheImageRef, err = metadataRetriever.GetCacheImage(cacheTag)
 		if err != nil {
-			return errors.Wrapf(err, "error obtaining home directory")
+			log.Fatal(err)
 		}
+	}
 
-		err = creds.Save(filepath.Join(homeDir, ".docker", "config.json"))
-		if err != nil {
-			return errors.Wrapf(err, "error writing docker creds")
+	builtImage, err := metadataRetriever.GetBuiltImage(builtImageRef)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	buildStatusMetadata := &build.BuildStatusMetadata{
+		BuildpackMetadata: buildMetadataFromBuiltImage(builtImage),
+		LatestImage:       builtImageRef,
+		LatestCacheImage:  cacheImageRef,
+		StackRunImage:     builtImage.Stack.RunImage,
+		StackID:           builtImage.Stack.ID,
+	}
+
+	compressor := build.GzipMetadataCompressor{}
+	data, err := compressor.Compress(buildStatusMetadata)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := ioutil.WriteFile(buildapi.CompletionTerminationMessagePath, []byte(data), 0666); err != nil {
+		log.Fatal(err)
+	}
+
+	if hasCosign() || notaryV1URL != "" {
```

Comment:
Because cosign uses the docker home keychain it can't use k8s crendential helpers. Correct?

Reply:
Nevermind. https://github.com/pivotal/kpack/pull/959/files#r875069531